alns/accept/LateAcceptanceHillClimbing.py

```diff
@@ -0,0 +1,73 @@
+from collections import deque
+
+from alns.accept.AcceptanceCriterion import AcceptanceCriterion
+
+
+class LateAcceptanceHillClimbing(AcceptanceCriterion):
+    """
+    The Late Acceptance Hill Climbing (LAHC) criterion accepts a candidate
+    solution when it is better than the current solution from a number of
+    iterations before.
+
+    This implementation is based on the description of LAHC in [1].
+
+    Parameters
+    ----------
+    history_length: int
+        Non-negative integer specifying the maximum number of previous solutions
+        to be stored. Default: 0 (i.e., no objectives stored).
+    greedy: bool
+        Bool indicating whether or not to accept candidate solutions if they are
+        better than the current solution. Default: False.
+    collect_better: bool
+        Bool indicating whether or not to only collect current solutions that
+        are better than the previous current. Default: False
```

Comment:
What do you think of this?

```python
"""
lookback_period: int
  Nonnegative integer specifying which solution to compare against
  for late acceptance. In particular, LAHC compares against the 
  then-current solution from `lookback_period` iterations ago. 
  Default 0, in which case LAHC reverts to regular hill climbing.
greedy: bool
  If set, LAHC always accepts a candidate that is better than the 
  current solution.
compare_current: bool
  If set, LAHC compares against the better of the current solution,
  and the solution from `lookback_period` iterations ago. If not,
  LAHC compares only against the solution from `lookback_period` 
  iterations ago.
"""
```

> Default 0,  in which case LAHC reverts to regular hill climbing.

Right?

Reply:
In light of the issue below, maybe not set a default for `lookback_period`.

Reply:
Works for me! This reads much better than the names from a few iterations ago :-).